controls/datagrid/filtering/programmatic-filtering.md

```diff
@@ -0,0 +1,217 @@
+                                           IsCaseSensitive="Falses"
+                                           Value="Barcelona"/>
+```
+
+### Distinct Values Filter Descriptor
+
+The `DistinctValuesFilterDescriptor` is a descriptor which filters by distinct values.
+
+It exposes the following properties:
+
+* `PropertyName`&mdash;Gets or sets the name of the property that is used to retrieve the filter value.
+* `Value`&mdash;Gets or sets the value used in the comparisons. This is the right operand of the comparison.
+
+<!-- <snippet id='datagrid-datetimefilterdescriptor-xaml'/> -->
+```XAML
+<telerik:DistinctValuesFilterDescriptor PropertyName="Country" Value="Austria" />
+```
+
+### Composite Filter Descriptor
+
+The `CompositeFilterDescriptor` represents a special `FilterDescriptorBase` that stores an arbitrary number of other descriptors instances. The logical `AND` or `OR` operator is applied upon all composed filters to determine the result of the `PassesFilter` routine.
+
+<!-- <snippet id='datagrid-compositefilterdescriptor-xaml'/> -->
+```XAML
+<telerik:CompositeFilterDescriptor Operator="And">
+	<telerik:CompositeFilterDescriptor.Descriptors>
+		<telerik:NumericalFilterDescriptor PropertyName="StadiumCapacity"
+                                           Operator="IsGreaterThan"
+                                           Value="55000"/>
+			<telerik:NumericalFilterDescriptor PropertyName="StadiumCapacity"
+                                               Operator="IsLessThan"
+                                               Value="85000"/>
+	</telerik:CompositeFilterDescriptor.Descriptors>
+</telerik:CompositeFilterDescriptor>
+```
+
+### Delegate Filter Descriptor
+
+The `DelegateFilterDescriptor` exposes the `Filter` property, which gets or sets the `IFilter` implementation used to check whether a data item passes the filter or not.
+
+To use a `DelegateFilterDescriptor`, you need to create a class that implements the `IFilter` interface which will return the `Key` by which you want to filter.
+
+Then, you need to add a `DelegateFilterDescriptor` to the `RadDataGrid.FilterDescriptors` collection and set its `Filter` property.
+
+The following example demonstrates the `CustomFilter` implementation:
+
+<!-- <snippet id='datagrid-delegatefilterdescriptor-csharp'/> -->
+```C#
+class CustomFilter : Telerik.Maui.Controls.Compatibility.Common.Data.IFilter
+{
+    public bool PassesFilter(object item)
+    {
+        if(item is Club club 
+           && club.StadiumCapacity > 60000 
+           && club.StadiumCapacity < 85000)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+}
+```
+
+> Take notice that `IFilter` is in `Telerik.Maui.Controls.Compatibility.Common.Data` namespace.
+
+Add the `DelegateFilterDescriptor` to the `RadDataGrid` instance:
+
+<!-- <snippet id='datagrid-delegatefilterdescriptor-added'/> -->
```

Comment:
replace the code snippets with a snippets from the sdk 
also mention the we have a demo in the sdk app.